File: CoreJava/SOLID-princples-app/src/dip/violation/TaxCalculator.java
package dip.violation;

public class TaxCalculator {
	private Logtype log;

	public TaxCalculator(Logtype log) {
		this.log = log;
	}

	public int calculateTax(int amount, int rate) {
		int result = 0;
		try {
			result = amount / rate;
		} catch (Exception ex) {
			if (log == log.XML)
				new XmlLogger().logError(ex.toString());
			else if (log == log.TXT)
				new TxtLogger().logError(ex.toString());
		}
		return result;
	}
}

File: DotNET/Solid Principles/DIPViolationApp/DIPViolationApp/TaxCalculator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace DIPViolationApp
{
    class TaxCalculator
    {
        private LogType _log;

        public TaxCalculator(LogType log)
        {
            _log = log;
        }

        public int CalculateTax(int amount, int rate)
        {
            int result = 0;
            try
            {
                result = amount / rate;
            }
            catch
            {
                if (_log == LogType.XML)
                    new XMLLogger().logError("Exception occured");
                else if (_log == LogType.TXT)
                    new TextLogger().logError("Exception occured");
            }
            return result;
        }
    }
}
